scotty

```diff
@@ -0,0 +1,458 @@
+CORE_DIR = os.path.dirname(os.path.realpath(__file__))
+Config = Config(CORE_DIR)
+
+
+def get_platform():
+    """ Detects the running platform, and if linux checks if a raspberry pi
+    """
+
+    plat = platform.system().lower()
+
+    if plat == 'linux' and platform.linux_distribution()[0].lower() == 'debian':
+        try:
+            with open('/proc/cpuinfo') as f:
+                for line in f:
+                    line = line.strip().split()
+                    if len(line) < 2:
+                        continue
+
+                    if line[0] == 'Hardware' and line[-1] in ['BCM2708', 'BCM2709', 'BCM2835']:
+                        return 'raspberry-pi'
+        except:
+            pass
+
+    return plat
+
+
+def get_file_path(f):
+    """ Returns the full file path of a file, inside the main project repo
+    """
+
+    return "%s/%s" % (CORE_DIR, f)
+
+
+def print_color(s, color="yellow"):
+    """Print with a color! Uses tput to output escape codes for colors
+    """
+
+    colors = ["black", "red", "green", "yellow", "blue", "magenta", "cyan", "white"]
+    color = color.lower()
+
+    if color in colors:
+        color_num = colors.index(color)
+        os.system("tput setaf %d 2>/dev/null" % color_num)
+        print s
+        os.system("tput sgr0 2>/dev/null")
+    else:
+        print s
+
+
+def set_env(env="base"):
+    """ Set environment variables, based on the config
+    """
+
+    if env not in Config.get("environment"):
+        print_color("\n%s environment set is not defined!" % env, color="red")
+        sys.exit(1)
+
+    env_set = Config.get("environment").get(env)
+
+    # These are paths, and should be appended to, not overwritten
+    path_set = ['PATH', 'LD_LIBRARY_PATH']
+
+    for key in env_set:
+        val = env_set[key].replace('${CORE_DIR}', CORE_DIR)
+
+        if key in path_set:
+            # If not there, just set our value
+            if key not in os.environ:
+                os.environ[key] = val
+
+            # Append it if not already in the path
+            elif val not in os.environ[key]:
+                os.environ[key] = val + ":" + os.environ[key]
+        else:
+            os.environ[key] = val
+
+
+def run_commands(commands):
+    """ Run a list of commands and print out what commands are being run when doing so
+    """
+
+    for com in commands:
+        print_color("\nRunning: %s \n" % com, color="cyan")
+        if os.system(com) != 0:
+            print_color("\nNon-zero error code running: %s\nStopping install." % com, color="red")
+            sys.exit()
+
+
+def create_venv():
+    """ Create a virtual env if it doesn't exist already
+    """
+
+    venv_real_dir = get_file_path(Config.get("venv"))
+
+    if not os.path.isdir(venv_real_dir):
+        print_color("\nCreating virtual env in %s..." % Config.get("venv"))
+        run_commands(["%s %s" % (str(Config.get("virtualenv")), str(venv_real_dir))])
```

Comment:
don't think you need `str(...)` here

Reply:
Didn't think so either, but it was showing unicode in the print statement it was doing something funky when I ran it without so that fixed it.